src/controllers/hid/hidcontroller.cpp

```diff
@@ -255,6 +255,35 @@ int HidController::close() {
     return 0;
 }
 
+QList<int> HidController::getInputReport(unsigned int reportID) {
+    Trace hidRead("HidController getInputReport");
+    unsigned char* pPreviousBuffer = m_pPollData[m_iPollingBufferIndex];
+    const int currentBufferIndex = (m_iPollingBufferIndex + 1) % kNumBuffers;
+    unsigned char* pCurrentBuffer = m_pPollData[currentBufferIndex];
+    int bytesRead;
+    pCurrentBuffer[0] = reportID;
+    bytesRead = hid_get_input_report(m_pHidDevice, pCurrentBuffer, kBufferSize);
+    controllerDebug(bytesRead << "bytes received by hid_get_input_report" << getName()
+                              << "serial #" << hid_serial
+                              << "(including report ID of" << reportID << ")");
+    bytesRead -= kReportIdSize;
```

Comment:
Error handing for bytesRead < 0 is missing.

Reply:
Done